classes/check/tasklatencycheck.php

```diff
@@ -53,89 +53,103 @@ public function get_result(): result {
         $lockstats = isset($CFG->lock_factory) && $CFG->lock_factory === \tool_lockstats\proxy_lock_factory::class;
 
         $tasks = explode(PHP_EOL, $taskconfig);
-        foreach ($tasks as $taskconfig) {
-            list($taskclass, $runtime, $startdelay, $completiondelay) = explode(',', str_replace(' ', '', $taskconfig));
-            $task = \core\task\manager::get_scheduled_task($taskclass);
 
-            // Input validation.
-            $valid = $task !== false;
+        foreach ($tasks as $taskconfig) {
+            if ($taskconfig) {
+                list($taskclass, $runtime, $startdelay, $completiondelay) = explode(',', str_replace(' ', '', $taskconfig));
```

Comment:
This is the main line throwing errors, so perhaps I shouldnt have assumed the count. So instead, here we could do the explode, count the elements = 4, throw the error if not, and if there is 4, do the  list assignment

```
$configarr = explode(',', str_replace(' ', '', $taskconfig));
if (count($configarr) !== 4) {
  return new result(result::ERROR, get_string('taskconfigbad', 'tool_heartbeat', $taskclass));
}
list($taskclass, $runtime, $startdelay, $completiondelay) = $configarr;
```